scripts/push_pull_run_element_info.py

```diff
@@ -0,0 +1,207 @@
+
+    def assess_sample(self, sample):
+        artifacts_to_upload = set()
+
+        artifacts = self.output_artifacts_per_sample(sample_name=sample.name)
+        un_reviewed_artifacts = [a for a in artifacts if a.udf.get('RE Review status') not in ['pass', 'fail']]
+        if un_reviewed_artifacts:
+            # Skip sample that have some un-reviewed run elements as they could still be sequencing and change the outcome of the review
+            return artifacts_to_upload
+
+        # Artifacts that pass the review
+        pass_artifacts = [a for a in artifacts if a.udf.get('RE Review status') == 'pass']
+
+        # Artifacts that fail the review
+        fail_artifacts = [a for a in artifacts if a.udf.get('RE Review status') == 'fail']
+
+        target_yield = float(sample.udf.get('Yield for Quoted Coverage (Gb)'))
+        good_re_yield = sum([float(a.udf.get('RE Yield Q30')) for a in pass_artifacts])
+
+        # Just the right amount of good yield: take it all
+        if target_yield < good_re_yield < target_yield * 2:
+            for a in pass_artifacts:
+                a.udf['RE Useable'] = 'yes'
+                a.udf['RE Useable Comment'] = 'AR: Good yield'
+            for a in fail_artifacts:
+                a.udf['RE Useable'] = 'no'
+                a.udf['RE Useable Comment'] = 'AR: Failed and not needed'
+            artifacts_to_upload.update(artifacts)
+
+        # Too much good yield limit to the best quality ones
+        elif good_re_yield > target_yield * 2:
+            # Too much yield: sort the good artifact by quality
+            pass_artifacts.sort(key=lambda x: x.udf.get('RE %Q30'), reverse=True)
+            current_yield = 0
+            for a in pass_artifacts:
+                current_yield += float(a.udf.get('RE Yield Q30'))
+                if current_yield < target_yield * 2:
+                    a.udf['RE Useable'] = 'yes'
+                    a.udf['RE Useable Comment'] = 'AR: Good yield'
+                else:
+                    a.udf['RE Useable'] = 'no'
+                    a.udf['RE Useable Comment'] = 'AR: To much good yield'
+            for a in fail_artifacts:
+                a.udf['RE Useable'] = 'no'
+                a.udf['RE Useable Comment'] = 'AR: Failed and not needed'
+            artifacts_to_upload.update(artifacts)
+
+        # Not enough good yield: manual decision
+        # Run element not passing review: manual decision
+
+        return artifacts_to_upload
+
+
+metrics_mapping_push = [
+    ('RE Useable', 'useable'),
+    ('RE Useable Comment', 'useable_comments'),
+]
+
+class PushRunElementInfo(RunElementInfo):
+
+    @cached_property
+    def current_time(self):
+        return datetime.datetime.now()
+
+    def _run(self):
+        # batch retrieve input and output artifacts along with samples
+        self.output_artifacts
+        for sample in self.samples:
+            run_elements = self.get_documents('aggregate/run_elements', match={'sample_id': sample.name})
```

Comment:
We only seem to be using the `useable` and `useable_comments` fields, both of which are in the schema, so we should be able to use `get_documents('run_elements', where=...`